src/main/scala/org/ergoplatform/local/ErgoMiner.scala

```diff
@@ -169,31 +170,38 @@ class ErgoMiner(ergoSettings: ErgoSettings,
     val timestamp = timeProvider.time()
     val stateContext = state.stateContext
     val nBits: Long = bestHeaderOpt
-      .map(parent => history.requiredDifficultyAfter(parent))
-      .map(d => RequiredDifficulty.encodeCompactBits(d))
+      .map(header => RequiredDifficulty.encodeCompactBits(history.requiredDifficultyAfter(header)))
       .getOrElse(Constants.InitialNBits)
-
-    val (extensionCandidate, votes: Array[Byte], version: Byte) = bestHeaderOpt.map { header =>
-      val newHeight = header.height + 1
-      val currentParams = stateContext.currentParameters
-      val betterVersion = protocolVersion > header.version
-      val votingFinishHeight: Option[Height] = currentParams.softForkStartingHeight
-        .map(h => h + votingSettings.votingLength * votingSettings.softForkEpochs)
-      val forkVotingAllowed = votingFinishHeight.forall(fh => newHeight < fh)
-      val forkOrdered = ergoSettings.votingTargets.getOrElse(Parameters.SoftFork, 0) != 0
-      val voteForFork = betterVersion && forkOrdered && forkVotingAllowed
-
-      if (newHeight % votingEpochLength == 0 && newHeight > 0) {
-        val newParams = currentParams.update(newHeight, voteForFork, stateContext.votingData.epochVotes, votingSettings)
-        (newParams.toExtensionCandidate(Seq()),
-          newParams.suggestVotes(ergoSettings.votingTargets, voteForFork),
-          newParams.blockVersion)
-      } else {
-        (ExtensionCandidate(Seq.empty),
-          currentParams.vote(ergoSettings.votingTargets, stateContext.votingData.epochVotes, voteForFork),
-          currentParams.blockVersion)
+    val interlinks = bestHeaderOpt
+      .flatMap { h =>
+        history.typedModifierById[Extension](h.extensionId)
+          .map(ext => updateInterlinks(h, unpackInterlinks(ext.fields)))
+      }
+      .getOrElse(Seq.empty)
+
+    val (extensionCandidate, votes: Array[Byte], version: Byte) = bestHeaderOpt
+      .map { header =>
```

Comment:
Looks like this does not corresponds to our style-guide (https://github.com/ergoplatform/ergo-scala-style-guide/blob/master/ErgoStyleGuide.md#14-calls-chaining), I'd prefer to have `.map` on the previous line

Reply:
@catena2w it did not fit into single line of 160 chars